autoload/probe/file.vim

```diff
@@ -72,6 +76,26 @@ function! probe#file#scan()
     return s:file_caches[s:hash]
 endfunction
 
+function! s:is_git()
+    let metadir_name = fnamemodify(s:find_metadir(), ':t')
+    return metadir_name == '.git'
+endfunction
+
+function! s:scan_git()
+    let files = []
+    echo "Scanning " . s:find_repo_root()
+
+    let cmd = "git ls-files; git submodule foreach -q 'for file in $(git ls-files); do echo \"$path/$file\"; done'"
```

Comment:
Having both `--cached` and `--other` is closest the `globpath` scan behaviour, returning all files that aren't explicitly ignored, so I was assuming that was the way to go.

hehe, if I recall correctly your workflows result in almost never have untracked files in your repo, but I speak for those undisciplined among us who do.